### DISCUSSION TITLE
Build scenery in a straight line

### DISCUSSION THREAD

**Discussion by PhyxionNL** [OpenRCT2/OpenRCT2#15667] [Feature requests] [answered]
With fences, it would be nice if we could 'generate' a line (e.g., by holding CTRL) to build a straight line between two points.

**Comment by IntelOrca** [comment]
This is planned for everything, e.g. paths, walls, roofs etc. and grid, box based too.

**Comment by KernelDeimos** [comment]
Has this been attempted by anyone yet? It looks like it's already half-done: the tile inspector can copy and paste elements and it knows about coordinates. Wouldn't it just be as simple as adding adding a "bulk paste" section to the tile inspector? I imagine three numeric inputs and a button, where the user can say "keep pasting the copied tile for <number of iterations> iterations while increasing X by <x increment> and Y by <y increment>"

I'd be willing to try this if nobody is already working on it

**Comment by ocalhoun6** [comment]
If you were going to put that much work into something like this, I think it would be much better and more versatile to be able to copy/paste large sections of terrain/scenery. Select a region (similar to how the cut-away view mode can select an area), then copy everything in it, and be able to paste it elsewhere, replacing whatever is in the new area. Maybe also with an option to change the height of everything you just pasted, in case you're pasting it at a different terrain level.

That could allow you to more efficiently build long lines of things (by making a short line, then pasting it several times), but it would also help do really useful things like duplicating buildings or parts of buildings, replicating the floors of buildings to easily make them taller, etc.

**Comment by KernelDeimos** [comment]
I think I have an idea of how to implement something like that. It would require a separate, more complicated copy&paste functionality than what's in the tile inspector, so I'll try to build something simpler first just to get the hang of things.

Who's familiar with the UI stuff? I'm probably going to have lots of questions.

Right now I'm trying to get the hang of the UI classes. I copied the spinner widgets for X and Y selection and made duplicates in `MAIN_TILE_INSPECTOR_WIDGETS` immediately after them, and then I added 6 more enum names in `WINDOW_TILE_INSPECTOR_WIDGET_IDX`. What I expected was that the behaviour of the tile inspector would not change in any way, but what actually happened was the existing spinners stopped working (maybe stopped listening to click events).

EDIT: Figured that out! The widgets I added were rendering over top of the other ones

**Comment by KernelDeimos** [comment]
Also kudos to whoever programmed this (the UI code & tile inspector). Before I added anything to the enum of widget identifiers, I expected the game would segfault or something. Instead, a whole bunch of totally ridiculous stuff happens, but the game still doesn't crash

**Comment by AaronVanGeffen** [comment]
Feel free to join us in [our Gitter chatroom](https://gitter.im/OpenRCT2/OpenRCT2) if you have any questions. That's a little more efficient than using the issue tracker as one. :)

**Comment by Ryder17z** [comment]
I like how fencing works in Zoo Tycoon - because it allows you to make L-shapes as well as straight lines.

It would be a pretty neat addition, but i don't know how difficult it would be to "copy" their function.

**Comment by ipatix** [comment]
Anyone still here interested in this? I'm currently having headache considering how painful building straight fences is compared to games like Zoo Tycoon.

**Comment by Gymnasiast** [comment, marked as answer]
Has been implemented a few releases back.
